libs/entry-components/configure-theming.md

```diff
@@ -0,0 +1,435 @@
+
+In summary, the generator:
+
+- Includes multiple generator files for supported entry components (dialog, table, button, form...)
+- Generates [Angular Material](https://material.angular.io/guide/theming) for customization of color pallets, typography, and density settings
+- Sets up  default, boilerplate theme with predefined values
+- Combines default theme and custom theme styles
+
+```scss
+@use 'entry-components/styles/generator';
+```
+
+## Create custom theme
+
+Next, define the $custom-theme map that will be expanded with various customization options based on specific needs.
+
+```scss
+$custom-theme: ();
+```
+
+Include and apply the styles generated by the generate-from mixin in the 'generator' module, using the customization options provided in the $custom-theme map.
+
+```scss
+@include generator.generate-from($custom-theme);
+```
+
+Our theme is now an empty object, and default library theme is used.
+
+```scss
+@use 'entry-components/styles/generator';
+
+$custom-theme: (
+);
+
+@include generator.generate-from($custom-theme);
+
+```
+
+## Importing Fonts
+
+There are 2 possibilities of using fonts in theming:
+
+### 1. Use default fonts from library
+
+**Entry-components** library comes with default fonts:  **Montserrat**, **OpenSans** and **Roboto** font family. To be able to use them, it needs to be configured in the `angular.json` file located at the root of the app. The fonts are imported from the library's node modules as follows:  
+
+```json
+"assets": [
+ "src/favicon.ico",
+ "src/assets",
+ {
+  "glob": "**/*",
+  "input": "./node_modules/@enigmatry/entry-components/assets/",
+  "output": "/assets/"
+ }
+],
+```
+
+### 2. Importing custom fonts
+
+Custom fonts needs to be imported locally in separated file using @font-face rule (our preferred SCSS structure is modules/components/typography/fonts.scss).
+
+```scss
+@font-face {
+ font: {
+  family: 'Helvetica';
+  weight: 400;
+ }
+ src: url('/assets/fonts/Helvetica.woff2') format('woff2'), url('/assets/fonts/Helvetica.woff') format('woff');
+}
+```
+
+## Theme Configuration Approaches
+
+We have 2 ways for configuring theme. We highly suggest using first approach, since it's very simple and easy to understand. But if there are special needs then native Angular Material approach is unavoidable.
+
+## 1. Simplified Configuration
+
+### Configuration Properties
+
+The new **custom-theme map**  should override the default theme that comes from a library. It contains a nested structure defining various aspects of a theme. Different key parameters can be passed in to extend it and customize it for our needs. The structure is organized into three main sections: **general**, **tables**, and **dialogs**. Each section contains specific configuration options.
```

Comment:
Whenever you create MD doc file, once done - use some online spell checker to polish everything.

Reply:
I already checked it, it passed without errors, so I will rephrase that paragraph.